pkg/codegen/go/gen.go

```diff
@@ -4883,6 +4888,32 @@ func GeneratePackage(tool string, pkg *schema.Package) (map[string][]byte, error
 		}
 	}
 
+	// create a go.mod file with references to local dependencies
+	if len(localDependencies) > 0 {
+		var vPath string
+		if pkg.Version != nil && pkg.Version.Major > 1 {
+			vPath = fmt.Sprintf("/v%d", pkg.Version.Major)
+		}
+
+		modulePath := fmt.Sprintf("github.com/pulumi/pulumi-%s/sdk%s", pkg.Name, vPath)
+		var gomod modfile.File
+		err = gomod.AddModuleStmt(modulePath)
+		contract.AssertNoErrorf(err, "could not add module statement to go.mod")
+		err = gomod.AddGoStmt("1.20")
+		contract.AssertNoErrorf(err, "could not add Go statement to go.mod")
+		pulumiPackagePath := "github.com/pulumi/pulumi/sdk/v3"
+		pulumiVersion := "v3.30.0"
+		err = gomod.AddRequire(pulumiPackagePath, pulumiVersion)
+		contract.AssertNoErrorf(err, "could not add require statement to go.mod")
+		if replacementPath, hasReplacement := localDependencies["pulumi"]; hasReplacement {
+			err = gomod.AddReplace(pulumiPackagePath, "", replacementPath, "")
+			contract.AssertNoErrorf(err, "could not add replace statement to go.mod")
+		}
+
+		files["go.mod"], err = gomod.Format()
```

Comment:
We should only emit this if "SupportPack" is enabled.

Reply:
Done ✅